src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,83 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final Integer BIT_SHIFT_BY_ONE = 1;
+    private Node[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        growIfSizeIsInLoadFactory();
+        Node newNode = new Node<>(key, value, null);
+        Node current = table[index];
+        while (current != null) {
+            if (checkKey(current, key)) {
+                current.value = value;
+                return;
+            }
+            current = current.next;
+        }
+        newNode.next = table[index];
+        table[index] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node current = table[getIndex(key)];
+        while (current != null) {
+            if (checkKey(current, key)) {
+                return (V) current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void growIfSizeIsInLoadFactory() {
```

Comment:
Please, consider better name for this method. Factory has nothing to do with this